apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.ExecutableCommand.cs

```diff
@@ -178,6 +179,38 @@ internal async Task<long> ExecutePartitionedUpdateAsync(CancellationToken cancel
                 return await transaction.ExecuteDmlAsync(request, cancellationToken, CommandTimeout).ConfigureAwait(false);
             }
 
+            internal async Task<string> GetInstanceEndpointUriAsync()
+            {
+                InstanceAdminClient instanceAdminClient =
+                    await InstanceAdminClient.CreateAsync().ConfigureAwait(false);
+                // Create request for instance with endpoints.
+                GetInstanceRequest request = new GetInstanceRequest
+                {
+                    InstanceName = new InstanceName(Connection.Project, Connection.SpannerInstance),
+                    FieldMask = new FieldMask { Paths = { "endpoint_uris" } }
+                };
+                try
+                {
+                    // Get list of available endpointUris from GetInstance.
+                    Instance instance = instanceAdminClient.GetInstance(request);
+                    return instance.EndpointUris.Count == 0 ? null : instance.EndpointUris[0];
```

Comment:
If there are more than one hosts, at the very least return one of them randomly.